### DISCUSSION TITLE
fast-xml 使用心得

### DISCUSSION THREAD

**Discussion by tidys** [NaturalIntelligence/fast-xml-parser#481] [Show and tell]
# 自闭和标签的注意事项
 我想要build result中hr为`<hr />`自闭和，使用 [suppressEmptyNode](https://github.com/NaturalIntelligence/fast-xml-parser/blob/master/docs/v4/3.XMLBuilder.md#suppressemptynode)可以实现，但是你需要注意的是，必须是没有`targetValue`
![image](https://user-images.githubusercontent.com/7894208/182775179-a90a76ad-22f5-4975-a61b-35acf5cf3214.png)

# innerHtml br的问题
 innertHtml有时会返回`<font size="20">1<br>2<br>3</font>`，没有自闭和<br/>
![image](https://user-images.githubusercontent.com/7894208/182777802-d69c9696-4be9-4ae6-9284-e20b99dc38ac.png)
直接parse就会导致将text解析到br标签里面
![image](https://user-images.githubusercontent.com/7894208/182778168-f6622e37-4f2b-4d82-b25f-2536f0d6240a.png)
如果在build之后，就会变为
`<font size="20">1<br>2</br><br>3</br></font>`
不符合预期，从源代码看，好像也没有好的解决方案，只能在parse之前`replace(/<br>/g, '<br/>');`

**Comment by amitguptagwl** [comment]
Sorry for the very late response. Have you checked [unpairedtags](https://github.com/NaturalIntelligence/fast-xml-parser/blob/master/docs/v4/2.XMLparseOptions.md#unpairedtags) property?

Let me know if that helps

**Reply by tidys** [reply]
It's been too long, I can't remember clearly anymore

**Comment by amitguptagwl** [comment]
Closing the discussion. Please reopen if need to discuss.
